hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/CallerContext.java

```diff
@@ -123,6 +140,37 @@ public Builder setSignature(byte[] signature) {
       return this;
     }
 
+    /**
+     * Append new item to the context.
+     * @param item
+     * @return builder
+     */
+    public Builder append(String item) {
+      if (isValid(item)) {
+        if (sb.length() > 0) {
+          sb.append(separator);
+        }
+        sb.append(item);
+      }
+      return this;
+    }
+
+    /**
+     * Append new item which contains key and value to the context.
+     * @param key
+     * @param value
+     * @return builder
+     */
+    public Builder append(String key, String value) {
+      if (isValid(key) && isValid(value)) {
+        if (sb.length() > 0) {
+          sb.append(separator);
+        }
+        sb.append(key + ":" + value);
```

Comment:
Should we make the colon a constant or even configurable?

Reply:
I was proposing fixed `:` for two reasons:
1. We have too many Hadoop configurations and I was thinking we add one more only when necessary. Using `:` seems most popular choice to me
1. There are not many choices I imagine and users have to know they can not use `\t`, `\n`, `=` etc. So I prefer simplicity. 

But I'm also fine if we go with a new config for this, +0 on that.

Thank you @goiri

Reply:
Let's make it at least a constant and not do + but appends.

Reply:
Yes making `:`a constant and replacing `+` with `append()` both make perfect sense.